generators/entity/prompts.js

```diff
@@ -276,6 +277,36 @@ function askForTableName() {
     });
 }
 
+function askForFiltering() {
+    // don't prompt if server is skipped, or the backend is not sql, or no service requested
+    if (this.useConfigurationFile || this.skipServer || this.databaseType !== 'sql' || this.service === 'no') {
+        return;
+    }
+    const done = this.async();
+    const prompts = [
+        {
+            type: 'list',
+            name: 'filtering',
+            message: 'Do you want to add filtering?',
+            choices: [
+                {
+                    value: 'no',
+                    name: 'Not needed'
+                },
+                {
+                    name: 'Dynamic filtering for the entities with JPA Static metamodel',
+                    value: 'jpaMetamodel'
+                }
+            ],
+            default: 0
+        }
+    ];
+    this.prompt(prompts).then((props) => {
+        this.filtering = props.filtering;
```

Comment:
I would like to make checking for stuff simpler, so may be can we also add a property called `this.jpaMetamodelFiltering = true` so that we can do `<% if (jpaMetamodelFiltering) {  %>` everywhere. I was thinking about refactoring the generator when I find time to follow the pattern of `this.options.elasticSearch`, `this.options.websocket` everywhere so that our EJS code becomes easier to maintain and refactor. This is how most other generators do options and I find it elegant.